core/amber/src/main/python/core/python_worker.py

```diff
@@ -31,12 +57,66 @@ def run(self) -> None:
         )
         main_loop_thread = Thread(target=self._main_loop.run, name="main_loop_thread")
 
+        stop_event = Event()
+        heartbeat_thread = Thread(
+            target=self.heartbeat,
+            name="heartbeat_thread",
+            args=(5.0, stop_event, network_sender_thread, main_loop_thread),
+        )
+
         network_sender_thread.start()
         main_loop_thread.start()
+        heartbeat_thread.start()
         main_loop_thread.join()
         network_sender_thread.join()
+        with self.condition:
+            self.condition.notify()
+
+        heartbeat_thread.join()
 
     @overrides
     def stop(self):
         self._main_loop.stop()
         self._network_sender.stop()
+        self_clean_child_process()
+
+    def heartbeat(self, interval, stop_event, network_sender_thread, main_loop_thread):
```

Comment:
Each thread in Python is written as a Runnable (for example, PythonWorker implements the Runnable protocol). Can you move this code into a separate Runnable definition?